src/ptf/testutils.py

```diff
@@ -1406,4 +1540,104 @@ def verify_packets_any(test, pkt, ports=[], device_number=0):
 
     test.assertTrue(received == True, "Did not receive pkt on any of ports %r for device %d" % (ports, device_number))
 
+def verify_any_packet_any_port(test, pkts=[], ports=[], device_number=0):
+    """
+    Check that _any_ of the packet is received on _any_ of the specified ports belonging to
+    the given device (default device_number is 0).
+
+    Also verifies that the packet is ot received on any other ports for this
+    device, and that no other packets are received on the device (unless --relax
+    is in effect).
+    """
+    received = False
+    index = 0
+    match_index = 0
+    (rcv_device, rcv_port, rcv_pkt, pkt_time) = test.dataplane.poll(
+        timeout=1,
+        filters=FILTERS
+    )
+    for port in ports:
+        if rcv_device != device_number:
```

Comment:
instead of having this test, the device_number should be passed as an argument to dp_poll